onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -454,74 +455,78 @@ std::vector<std::unique_ptr<ComputeCapability>> OpenVINOExecutionProvider::GetCa
 
   std::set<const onnxruntime::NodeArg*> fused_inputs, fused_outputs;
 
-  if (IsGraphSupported(graph_viewer,device_id)) {
-    std::string model_proto_strbuf;
-    model_proto.SerializeToString(&model_proto_strbuf);
+  if (!IsGraphSupported(graph_viewer,device_id)) {
+    LOGS_DEFAULT(WARNING) << openvino_ep::OpenVINOGraph::log_tag << "Rejecting as graph has unsupported operations.";
+    return result;
+  }
 
-    std::string xml_string, weights_string;
+  std::string model_proto_strbuf;
+  model_proto.SerializeToString(&model_proto_strbuf);
 
-    // Try converting with OpenVINO's Model Optimizer
-    try {
-      openvino_ep::OpenVINOGraph::ConvertONNXModelToOpenVINOIR(model_proto_strbuf, xml_string, weights_string, precision_fp32);
-    } catch  (const char* msg) {
-      // Model Optimizer cannot convert this model.
-      return result;
-    }
+  std::string xml_string, weights_string;
 
-    auto node_indexes = graph_viewer.GetNodesInTopologicalOrder();
+  // Try converting with OpenVINO's Model Optimizer
+  try {
+    openvino_ep::OpenVINOGraph::ConvertONNXModelToOpenVINOIR(model_proto_strbuf, xml_string, weights_string, precision_fp32);
+  } catch  (const char* msg) {
+    // Model Optimizer cannot convert this model.
+    LOGS_DEFAULT(WARNING) << openvino_ep::OpenVINOGraph::log_tag << "Rejecting as Model Optimizer cannot convert this model.";
+    return result;
```

Comment:
Should msg be logged as well? Assuming it describes why it cannot be converted which would be helpful for the user to know.

Reply:
Yes. msg should be logged as well.